providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -211,10 +198,6 @@ private void onProviderEvent(FlagdProviderEvent flagdProviderEvent) {
 
         synchronized (eventsLock) {
             log.info("FlagdProviderEvent: {}", flagdProviderEvent.getEvent());
-            eventsLock.syncMetadata = flagdProviderEvent.getSyncMetadata();
```

Comment:
This was the bug. We only care about this on READY events. I've mode it and added a comment: https://github.com/open-feature/java-sdk-contrib/pull/1226/files#diff-7ac82836e0450dd1f6581536670fba629b6fa21a8ccb4a5ffcdce54a53dd57e3R219-R222